### PR TITLE
Search and expand

### PR DESCRIPTION
**Pull request by barbarah**

### CODE REVIEW

File: src/app/[locale]/filter-set.tsx [hunk 1]
```diff
@@ -10,14 +19,61 @@ interface Props {
...
+              filterName: title.toLocaleLowerCase(),
+            })}
+            value={searchText}
+            onChange={e => setSearchText(e.target.value)}
+            type="text"
+            className="block rounded-md grow border-gray-300 px-4 py-1 shadow-sm focus:border-indigo-500 focus:ring-indigo-500 sm:text-sm"
+          ></input>
+          <button
+            className="ml-3 inline-flex items-center text-blue-link"
+            onClick={() => setIsExpanded((bool: boolean): boolean => !bool)}
```
Comment (sdevalk) [inline src/app/[locale]/filter-set.tsx:60]: Perhaps make the name `bool` more descriptive? E.g. `isExpanded`?

File: src/app/[locale]/filter-set.tsx [hunk 2]
```diff
@@ -10,14 +19,61 @@ interface Props {
...
+  Props,
+  'title' | 'selectedFilters' | 'setSelectedFilters'
+> & {
+  searchResultFilter: SearchResultFilter;
+};
+
+type HeaderProps = Pick<Props, 'title' | 'searchResultFilters'> & {
+  setIsExpanded: Dispatch<React.SetStateAction<boolean>>;
+  setSearchText: Dispatch<React.SetStateAction<string>>;
+  searchText: string;
```
Comment (sdevalk) [inline src/app/[locale]/filter-set.tsx:32]: We could harmonize the names of these properties?

https://github.com/colonial-heritage/dataset-browser/blob/main/src/app/%5Blocale%5D/client-filters.tsx#L38 uses `query` and `setQuery`.

File: src/app/[locale]/filter-set.tsx [hunk 3]
```diff
@@ -31,38 +87,152 @@ export default function FilterSet({
...
+  const {getReferenceProps, getFloatingProps} = useInteractions([
+    useDismiss(context),
+  ]);
+  const [searchText, setSearchText] = useState('');
+  const filteredSearchResultFilters: SearchResultFilter[] = useMemo(() => {
+    return searchResultFilters.filter(searchResultFilter =>
+      searchResultFilter.name
+        .toLocaleLowerCase()
+        .includes(searchText.toLocaleLowerCase())
+    );
```
[the remark this answers is not in the document]
Comment (barbarah) [reply src/app/[locale]/filter-set.tsx]: Good point. I will do this separately.

[the remark this answers is not in the document]
Comment (sdevalk) [reply src/app/[locale]/filter-set.tsx]: 👍🏻

**Review by sdevalk — APPROVED** [review]
👍🏻

**Pull request merged**